lib/hydra/derivatives/processors/active_encode.rb

```diff
@@ -1,25 +1,24 @@
 require 'active_encode'
 
 module Hydra::Derivatives::Processors
+  class ActiveEncodeError < StandardError
+    def initialize(status, source_path, errors = [])
+      msg = "ActiveEncode status was \"#{status}\" for #{source_path}"
+      msg = "#{msg}: #{errors.join(' ; ')}" unless errors.empty?
+      super(msg)
+    end
+  end
+
   class ActiveEncode < Processor
     def process
       encode = ::ActiveEncode::Base.create(source_path, directives)
+
+      # Wait until the encoding job is finished
       # while(encode.reload.running?) { sleep 10 }
 
-      raise_exception_if_encoding_failed(encode)
-      raise_exception_if_encoding_cancelled(encode)
+      raise ActiveEncodeError.new(encode.state, source_path, encode.errors) unless encode.completed?
```

Comment:
Will this always throw an error since we're not blocking until the job is finished yet?  Seems like it will always have a status of :running when it gets to this point.